hologram_dashboard/server/request_handler/request_handler_test.cc

```diff
@@ -0,0 +1,100 @@
+const std::string expected_chipper_GDPR_value = R"json({"ANDROID_SPAM_ABUSE":[["5/19/2020",true],["5/18/2020",true],["5/17/2020",true],["5/16/2020",true],["5/15/2020",true],["5/14/2020",true],["5/13/2020",true]],"APPS_DAILY_DATA_SOURCE":[["5/19/2020",true],["5/18/2020",true],["5/17/2020",true],["5/16/2020",true],["5/15/2020",false],["5/14/2020",false],["5/13/2020",false]],"APPS_TITLE_TIER":[["5/19/2020",true],["5/18/2020",false],["5/17/2020",true],["5/16/2020",true],["5/15/2020",true],["5/14/2020",false],["5/13/2020",false]],"APP_COUNTRY_PUBLISH_TIME":[["5/19/2020",false],["5/18/2020",false],["5/17/2020",true],["5/16/2020",false],["5/15/2020",false],["5/14/2020",true],["5/13/2020",true]],"APP_TYPE_SOURCE":[["5/19/2020",true],["5/18/2020",true],["5/17/2020",true],["5/16/2020",false],["5/15/2020",true],["5/14/2020",false],["5/13/2020",true]],"DEVICE_INTEGRITY_SOURCE":[["5/19/2020",true],["5/18/2020",false],["5/17/2020",false],["5/16/2020",true],["5/15/2020",true],["5/14/2020",true],["5/13/2020",false]],"PLAY_COUNTRY":[["5/19/2020",false],["5/18/2020",false],["5/17/2020",false],["5/16/2020",true],["5/15/2020",false],["5/14/2020",true],["5/13/2020",false]],"PRIME_MERIDIAN":[["5/19/2020",true],["5/18/2020",true],["5/17/2020",true],["5/16/2020",true],["5/15/2020",true],["5/14/2020",true],["5/13/2020",true]],"QUERY_CATEGORY_SOURCE":[["5/19/2020",false],["5/18/2020",false],["5/17/2020",false],["5/16/2020",false],["5/15/2020",true],["5/14/2020",false],["5/13/2020",true]],"SHIELD_ID":[["5/19/2020",false],["5/18/2020",false],["5/17/2020",true],["5/16/2020",true],["5/15/2020",true],["5/14/2020",false],["5/13/2020",true]],"SPAM":[["5/19/2020",false],["5/18/2020",true],["5/17/2020",true],["5/16/2020",true],["5/15/2020",true],["5/14/2020",true],["5/13/2020",true]],"USER_ATTRIBUTE":[["5/19/2020",true],["5/18/2020",true],["5/17/2020",true],["5/16/2020",true],["5/15/2020",true],["5/14/2020",true],["5/13/2020",true]],"USER_EXPERIMENT_ID_AND_TEST_CODE":[["5/19/2020",true],["5/18/2020",true],["5/17/2020",true],["5/16/2020",true],["5/15/2020",true],["5/14/2020",true],["5/13/2020",true]]})json";
+const std::string expected_mock_injection_command = 
+    "sample command to run backfill for SPAM on Chipper for 08-07-2020";
```

Comment:
I feel these expected results are not always as expected. If the server returns you something else, then the test would break, right? The server is supposed to read from DB where the date and availability status should change over time. In this case, the test is fragile.

Plus, I still think that using the client is more like the e2e test, not unit test.

Reply:
Actually after a second thought, perhaps the thing we need to fix is to override the data which the request handler get fetch in the test.

One way is to move the data fetching logic to a separate class (e.g. DataAvailabilityReader), and then mock DataAvailabilityReader to return the data you want in the test. This way we could ensure the expected result is static.